### PR TITLE
Add nightly-test-runner profile

### PR DESCRIPTION
**Pull request by peeech**

### CODE REVIEW

**Comment by alastairong** [comment]
@peeech Is this still needed?

File: profiles/logical/holo/nightly-test-runner/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  networking.firewall.allowedTCPPorts = [ 80 443 ];
+  networking.firewall.allowPing = true;
+
+  environment.systemPackages = [
+    git
+    nodejs
+    jq
+    yarn
+    brave
```
Comment (alastairong) [inline profiles/logical/holo/nightly-test-runner/default.nix:24]: @peeech If this is the browser we should not be prioritising it, but rather Chrome at this point. Just a reality of the market.

Comment (peeech) [reply profiles/logical/holo/nightly-test-runner/default.nix]: `brave == chrome == chromium`
With exception that in nixOs some behave better in a headless mode, some worse.

Comment (alastairong) [reply profiles/logical/holo/nightly-test-runner/default.nix]: You mean `==` for the purposes of headless testing?

File: modules/services/nightly-test-runner.nix [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    # ssh key for ssh-ing into test holoports
+    export SSH_KEY_PATH=${cfg.working-directory}/.ssh/id_ed25519
+    # github key for accessing private repo nightly-network-test
+    GITHUB_KEY=$(cat keys/github_key)
+    # test result file name
+    RESULT_FILE=nightly-test-result-$(date +%F).txt
+    # mattermost details, where test result file will be posted
+    MM_URL="https://chat.holochain.org"
+    MM_TOKEN=$(cat keys/mattermost_key)
+    MM_CHANNEL_ID=cq16udayciboin8oyk8xb1ofkr
```
Comment (zo-el) [inline modules/services/nightly-test-runner.nix:23]: Again, not a blocker but I noticed that the results we posted in the private channel, weren't we going to post it on the public channel?

**Review by zo-el — APPROVED** [review]
Looks great. just left one small question.

File: modules/services/nightly-test-runner.nix (hunk 2 above)
Comment (peeech) [reply modules/services/nightly-test-runner.nix:23]: I updated code to post to public channel

File: modules/services/nightly-test-runner.nix [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+  config = mkIf (cfg.enable) {
+    systemd.services.nightly-test-runner = {
+      after = [ "network.target" "zerotierone.service" ];
+      requisite = [ "zerotierone.service" ];
+      wantedBy = [ "multi-user.target" ];
+      restartIfChanged = false;
+      startAt = "*-*-* 06:15";
+
+      environment.PUPPETEER_EXECUTABLE_PATH = "${google-chrome}/bin/google-chrome-stable";
```
[the remark this answers is not in the document]
Comment (peeech) [reply modules/services/nightly-test-runner.nix:101]: great :-)

**Review by JettTech — APPROVED** [review]
LGTM.  I would like to request that you make sure that the frequency interval with which the results are posted to the mm channel is correct -  but otherwise great additions! :tada:

**Pull request merged**